### PR TITLE
UI fixes for permissions

### PR DESCRIPTION
**Pull request by Tiggles**

### CODE REVIEW

**Comment by Tiggles** [comment]
`docker: { peer?: boolean }` still needs to be handled

File: frontend-web/webclient/app/Applications/Jobs/View.tsx [hunk 1]
```diff
@@ -481,12 +492,17 @@ const InQueueText: React.FunctionComponent<{job: Job}> = ({job}) => {
 
     useEffect(() => {
         const support = job.status.resolvedSupport?.support as ComputeSupport;
-        if (support.docker.utilization === true || support.virtualMachine.utilization === true) {
+        const appType = getAppType(job);
+        if (
+            (appType === "DOCKER" && support.docker.utilization === true) ||
+            (appType === "VIRTUAL_MACHINE" && support.virtualMachine.utilization === true)
+        ) {
```
Comment (Tiggles) [inline frontend-web/webclient/app/Applications/Jobs/View.tsx:499]: I added this check as I believe it was missing in the old approach.

File: frontend-web/webclient/app/Applications/Jobs/View.tsx [hunk 2]
```diff
@@ -481,12 +492,17 @@ const InQueueText: React.FunctionComponent<{job: Job}> = ({job}) => {
...
             setUtilization(compute.jobs.retrieveUtilization({jobId: job.id}))
         }
     }, [job]);
 
     return <>
+        {/* Note(Jonas): Line below is outdated, right? */}
         <Heading.h2>{CONF.PRODUCT_NAME} is preparing your job</Heading.h2>
```
Comment (Tiggles) [inline frontend-web/webclient/app/Applications/Jobs/View.tsx:506]: Should this be based on the provider instead?

File: frontend-web/webclient/app/Applications/Jobs/View.tsx [hunk 3]
```diff
@@ -565,6 +581,21 @@ const InfoCardsContainer = styled.div`
...
+            return support?.virtualMachine[flag] === true;
+        case "NATIVE":
+            return support?.native[flag] === true;
+        default: {
+            console.log(`Unhandled job backend: ${jobBackend}`)
+            return false;
+        }
+    }
+}
+
```
Comment (Tiggles) [inline frontend-web/webclient/app/Applications/Jobs/View.tsx:598]: SINGULARITY always defaults to `false`.

File: frontend-web/webclient/app/Applications/Jobs/View.tsx (hunk 2 above)
Comment (DanThrane) [reply frontend-web/webclient/app/Applications/Jobs/View.tsx:513]: Yes, I think it should say something different. This is probably the simplest solution.

```suggestion
        <Heading.h2>Your job is being prepared</Heading.h2>
```

File: frontend-web/webclient/app/Resource/Properties.tsx [hunk 4]
```diff
@@ -382,3 +386,13 @@ const Messages: React.FunctionComponent<{resource: Resource}> = ({resource}) =>
...
     return <Box height={"200px"} ref={termRef} />
 };
+
+function canEditPermission(support: ProductSupport | undefined, namespace: string): boolean {
+    switch (namespace) {
+        case "files.collections":
+            return !!(support?.["collection"]?.["aclModifiable"]);
+        case "files":
+            return !!(support?.["files"]?.["aclModifiable"]);
+        default: return false;
```
Comment (DanThrane) [inline frontend-web/webclient/app/Resource/Properties.tsx:396]: Default should be true not false.

File: frontend-web/webclient/app/UCloud/FileCollectionsApi.tsx [hunk 5]
```diff
@@ -63,6 +63,8 @@ export interface FileCollectionSupport extends ProductSupport {
         aclSupported?: boolean;
         aclModifiable?: boolean;
 
+        shareSupport?: boolean;
```
Comment (DanThrane) [inline frontend-web/webclient/app/UCloud/FileCollectionsApi.tsx:66]: Could you create an issue about adding backend support for this?

**Review by DanThrane — COMMENTED** [review]
A quick look and a few comments.

File: frontend-web/webclient/app/UCloud/FileCollectionsApi.tsx (hunk 5 above)
Comment (Tiggles) [reply frontend-web/webclient/app/UCloud/FileCollectionsApi.tsx:66]: Yes

**Review by DanThrane — APPROVED** [review]

**Pull request merged**